src/go/rpk/pkg/cli/cmd/redpanda/config.go

```diff
@@ -131,7 +131,10 @@ func bootstrap(fs afero.Fs) *cobra.Command {
 			ownIP, err := parseSelfIP(self)
 			out.MaybeDieErr(err)
 
-			cfg.Redpanda.ID = id
+			if id >= 0 {
+				cfg.Redpanda.ID = new(int)
+				*cfg.Redpanda.ID = id
```

Comment:
These two lines can be squashed into `cfg.Redpanda.ID = &id`

Reply:
Done